pkg/output/stdout.go

```diff
@@ -431,6 +433,11 @@ func (s *Stdoutput) resultNormal(res ffuf.Result) {
 	fmt.Println(resnormal)
 }
 
+func (s *Stdoutput) resultJson(res ffuf.Result) {
+	resBytes, err := json.Marshal(res)
+	if err != nil {s.Error(err.Error())} else {fmt.Println(string(resBytes))}
```

Comment:
I think you could mitigate the stderr status bar issues here by doing something like:

```
fmt.Printf("%s%s\n", TERMINAL_CLEAR_LINE, string(resBytes))
```

Reply:
I don't love this, it introduces cruft into STDOUT which would likely mess up the consumer of JSON output

```
% ./ffuf -u https://FUZZ1/FUZZ2 -w ~/domains.txt:FUZZ1 -w ~/wordlist.txt:FUZZ2 -json -H "User-Agent: Mozilla" 2>/dev/null | head -c40 | xxd
00000000: 0d1b 5b32 4b7b 2269 6e70 7574 223a 7b22  ..[2K{"input":{"
00000010: 4655 5a5a 3122 3a22 6433 6433 4c6d 6868  FUZZ1":"d3d3Lmhh
00000020: 5932 746c 636d 3975                      Y2tlcm9u
```

How about this:

```
func (s *Stdoutput) resultJson(res ffuf.Result) {
	resBytes, err := json.Marshal(res)
	if err != nil {
		s.Error(err.Error())
	} else {
		fmt.Fprint(os.Stderr, TERMINAL_CLEAR_LINE)
		fmt.Println(string(resBytes))
	}
}
```

```
% ./ffuf -u https://FUZZ1/FUZZ2 -w ~/domains.txt:FUZZ1 -w ~/wordlist.txt:FUZZ2 -json -H "User-Agent: Mozilla"

[... SNIP - banner removed ...]
________________________________________________

{"input":{"FUZZ1":"d3d3LmhhY2tlcm9uZS5jb20=","FUZZ2":"aW5kZXgucGhw"},"position":4,"status":301,"length":31884,"words":570,"lines":12,"content-type":"text/html; charset=UTF-8","redirectlocation":"https://www.hackerone.com/","url":"https://www.hackerone.com/index.php","duration":36923533,"resultfile":"","host":"www.hackerone.com"}
{"input":{"FUZZ1":"d3d3LmhhY2tlcm9uZS5jb20=","FUZZ2":"cm9ib3RzLnR4dA=="},"position":6,"status":200,"length":1586,"words":129,"lines":62,"content-type":"text/plain","redirectlocation":"","url":"https://www.hackerone.com/robots.txt","duration":42903359,"resultfile":"","host":"www.hackerone.com"}
{"input":{"FUZZ1":"YnVnY3Jvd2QuY29t","FUZZ2":"cm9ib3RzLnR4dA=="},"position":5,"status":200,"length":53,"words":4,"lines":5,"content-type":"text/plain; charset=UTF-8","redirectlocation":"","url":"https://bugcrowd.com/robots.txt","duration":41099455,"resultfile":"","host":"bugcrowd.com"}
:: Progress: [6/6] :: Job [1/1] :: 74 req/sec :: Duration: [0:00:01] :: Errors: 0 ::

% ./ffuf -u https://FUZZ1/FUZZ2 -w ~/domains.txt:FUZZ1 -w ~/wordlist.txt:FUZZ2 -json -H "User-Agent: Mozilla" 2>/dev/null | head -c40 | xxd
00000000: 7b22 696e 7075 7422 3a7b 2246 555a 5a31  {"input":{"FUZZ1
00000010: 223a 2264 3364 334c 6d68 6859 3274 6c63  ":"d3d3LmhhY2tlc
00000020: 6d39 755a 5335 6a62                      m9uZS5jb
```

Note that this is giving me an IDE warning for ignoring the `error` from the first `fmt.Fprint()` but we don't seem to be handling these from any other print statement anyway. **That won't be a problem right?** (I'm still new enough to Go to be unsure about warnings such as these)

Finally, this will always emit a "clear line" control sequence to STDERR per result.

1. If there is currently an option to suppress the progress lines, this means we'll be spitting unnecessary and potentially dangerous "clear line" control sequences to STDERR per result. **We don't currently have such an option do we?**
2. If, in future, an option is added to suppress the progress lines, someone will need to make the printing of the "clear line" control sequence in this block of code conditional on that suppression option, else we'll fall victim to item 1. **Are we happy to kick that can down that road?**

Reply:
I agree with every concern you have here. 

I think the most reasonable thing to do is to add a new function for emitting TERM_CLEAR_LINE to STDERR, and call that in every occassion where TERM_CLEAR_LINE is currently prepended to STDOUT. If in the future the progress bar gets disabled, we can check that flag in that newly created function to fix it once and for all STDOUT prints.

That said, I don't let that architectural change block this PR here, but I'll do it later on.

Reply:
> Note that this is giving me an IDE warning for ignoring the error from the first fmt.Fprint() but we don't seem to be  handling these from any other print statement anyway. That won't be a problem right? (I'm still new enough to Go to  be unsure about warnings such as these)

Yeah, it's all good